### DISCUSSION TITLE
Observation Import Performance

### DISCUSSION THREAD

**Discussion by jutglar** [FraunhoferIOSB/FROST-Server#1563] [Q&A] [answered]
We import around 0.5-1million observations per datastream at once using dataArray.
 [OGC SensorThings API Part 1: Sensing Version 1.1 (opengeospatial.org)](https://docs.opengeospatial.org/is/18-088/18-088.html#create-observation-dataarray)
Lately, we received an empty response from server by importing 0.6 million observations. 
Do you suggest other request than data Array for importing a lot of observations?
Do you suggest splitting observations import in multiple POST? What max size should have an observation POST using dataArray?

Thank you for the support!

**Comment by hylkevds** [comment, marked as answer]
I usually initially limit the number of items in a DataArray to 1000, but do experiment with different amounts to find the optimum for your setup. If processing a single request takes a long time you may run into timeouts.

**Reply by jutglar** [reply]
Thanks for the advice. We decieded to select 50000 observations per DataArray, having 1min POST runtime on our local machine.
